backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/Boxing.kt

```diff
@@ -61,7 +60,7 @@ internal val Context.getBoxFunction: (IrClass) -> IrSimpleFunction by Context.la
             startOffset, endOffset,
             DECLARATION_ORIGIN_INLINE_CLASS_SPECIAL_FUNCTION,
             IrSimpleFunctionSymbolImpl(),
-            Name.special("<${inlinedClass.name}-box>"),
+            Name.special("<${inlinedClass.fqNameForIrSerialization}-box>"),
```

Comment:
It makes some sense to make this less verbose, by using a "relative" class name (i.e. "fq name" that doesn't include package). This also might help to implement a missing pre-condition ([see above](https://github.com/JetBrains/kotlin-native/pull/4733/files#r586175920)), by extracting something like `IrClass.relativeNameFromTopLevel: String?` or whatever.